engine/action_contact.go

```diff
@@ -103,6 +96,85 @@ func (c *Client) AddContactRequest(ctx context.Context, fullName, paymailAdress,
 	return contact, nil
 }
 
+func (c *Client) GetContacts(ctx context.Context, xPubID string, metadata *Metadata, conditions map[string]interface{}, queryParams *datastore.QueryParams) ([]*Contact, error) {
+	contacts, err := getContacts(ctx, xPubID, metadata, conditions, queryParams, c.DefaultModelOptions()...)
+	if err != nil {
+		return nil, err
+	}
+
+	return contacts, nil
+}
+
+func (c *Client) AcceptContact(ctx context.Context, xPubID, paymail string) error {
+
+	contact, err := getContact(ctx, paymail, xPubID, c.DefaultModelOptions()...)
+	if err != nil {
+		c.logContactError(xPubID, paymail, fmt.Sprintf("unexpected error while geting contact: %s", err.Error()))
+		return err
+	}
+	if contact == nil {
+		return ErrContactNotFound
+	}
+
+	if err = contact.Accept(); err != nil {
+		c.logContactWarining(xPubID, paymail, err.Error())
+		return ErrContactIncorrectStatus
+	}
+
+	if err = contact.Save(ctx); err != nil {
+		c.logContactError(xPubID, paymail, fmt.Sprintf("unexpected error while saving contact: %s", err.Error()))
```

Comment:
The phrase `unexpected error while saving contact` is repeated multiple times. As far as I see it, the phrase could be moved to the `logContactError` method.

Reply:
I currently use `logContactError` in two scenarios: for errors encountered while reading from the database and for errors occurring during writing to the database. I don't see any possibility to consolidate these messages into a single method, other than creating separate log methods for each situation. However, I would prefer to avoid this approach as it may result in less readable code in my opinion

Reply:
Agreed 👍